### PR TITLE
Added extra check for Accept header on batch secrets request

### PR DESCRIPTION
**Pull request by telday**
This check makes sure that the header exists before
attempting to check its value in order to prevent errors

### What does this PR do?

Added an extra check to make sure the `Accept` header is actually included in a request before trying to compare its value to anything. If the header is totally absent Conjur will throw a 500 error. The reason that tests have not found this issue yet is because Ruby's `RestClient::` [implicitly adds](https://github.com/rest-client/rest-client/blob/master/lib/restclient/request.rb#L588) an `Accept: */*` header to all requests if another value is not specified. However other languages do not do this.

After some digging it does not seem like there is an easy way to override the default headers set in `RestClient::` and `Net::` so testing this functionality might be difficult. Due to the impending OSS suite release I think this change should be pulled in and a ticket opened to write a new test for this edge case.

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: cucumber/api/features/support/rest_helpers.rb [hunk 1]
```diff
@@ -49,6 +52,30 @@ def get_json(path, options = {})
...
+    uri, headers = URI(args[0] + path), args[1][:headers]
+    http = Net::HTTP.new(uri.host, uri.port)
+
+    request = Net::HTTP::Get.new(uri.request_uri)
+    headers.each do |key, value|
+      request[key] = value
+    end
+
+    if options[:user] && options[:password]
+      request.basic_auth options[:user], options[:password]
```
Comment (jonahx) [inline cucumber/api/features/support/rest_helpers.rb:68]: Add parens for method call.

File: cucumber/api/features/support/rest_helpers.rb [hunk 2]
```diff
@@ -49,6 +52,30 @@ def get_json(path, options = {})
     set_result result
   end
 
+  def get_json_raw(path, options = {})
+    path = denormalize(path)
+    args = get_request_args options
```
Comment (jonahx) [inline cucumber/api/features/support/rest_helpers.rb:57]: Add parens for method call.

File: cucumber/api/features/support/rest_helpers.rb [hunk 3]
```diff
@@ -339,6 +366,12 @@ def rest_resource options
     args.last[:headers] ||= {}
     args.last[:headers].merge(headers) if headers
 
+    args
+  end
+
+  def rest_resource options
+    args = get_request_args options
```
Comment (jonahx) [inline cucumber/api/features/support/rest_helpers.rb:373]: parens

File: cucumber/api/features/support/rest_helpers.rb [hunk 4]
```diff
@@ -49,6 +52,32 @@ def get_json(path, options = {})
     set_result result
   end
 
+  def get_json_no_default_headers(path, options = {})
```
Comment (jonahx) [inline cucumber/api/features/support/rest_helpers.rb:55]: We should a comment above this function explaining the reason we're using `Net::HTTP` instead of `RestClient`.  Something like:

```
# Since there is no way to remove the default Accept header in RestClient
# we use Net::HTTP here.  Otherwise, we wouldn't be able to simulate requests
# that omit the Accept header in our tests.
```

**Review by jonahx — APPROVED** [review]
